includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1807,19 +1807,12 @@ private function replace_node_with_children( $node ) {
 			return;
 		}
 
-		/*
-		 * If node has children, replace it with them and push children onto stack
-		 *
-		 * Create a DOM fragment to hold the children
-		 */
-		$fragment = $this->dom->createDocumentFragment();
-
 		// Add all children to fragment/stack.
-		$child = $node->firstChild;
-		while ( $child ) {
-			$fragment->appendChild( $child );
-			$this->stack[] = $child;
-			$child         = $node->firstChild;
+		$children = array();
+		foreach ( $node->childNodes as $child ) {
+			$child->originalParent = $node->nodeName; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.NotSnakeCaseMemberVar
```

Comment:
Without keeping track of the original parent somehow, children of the stripped element will have a different parent reported.

For example, when this is validated:
```html
<div><invalid-element><inner-invalid></inner-invalid></invalid-element></div>
```

This wil strip `<invalid-element>`:
```html
<div><inner-invalid></inner-invalid></div>
```

So `<inner-invalid>` will have a reported parent of `<div>`, even though its original parent was `<invalid-element>`.